### PR TITLE
[cfg] Update labels file for updated checkers

### PR DESCRIPTION
**Pull request by gamesh411**
Since the last release some ClangSA checkers were moved to different packages and some new checkers were introduced. This change updates the label configuration of CodeChecker to reflect this.

I just eyeballed the severities and the profiles, so it's open for suggestions.

Notes:
alpha.security.taint.TaintPropagation does not have a documentation link, so I just put the GenericTaintChecker link as doc-url. Alternatively, we could use doc entry: https://clang.llvm.org/docs/analyzer/user-docs/TaintAnalysisConfiguration.html

With this patch, all the taint-related checkers share the same severity and profiles as the now-removed alpha.security.taint.TaintPropagation.

I put BlockInCriticalSection in the default profile, as I have deemed it not too noisy, and the results were meaningful IMO.

### CODE REVIEW

File: config/labels/analyzers/clang-tidy.json [hunk 1]
```diff
@@ -194,6 +198,10 @@
       "sei-cert:pos44-c",
       "severity:MEDIUM"
     ],
+    "bugprone-bitwise-pointer-cast": [
```
Comment (dkrupp) [inline config/labels/analyzers/clang-tidy.json:201]: please consider adding all bugprone chekcers to the default profile if they are not too noisy on the open source projects.

File: config/labels/analyzers/clangsa.json [hunk 2]
```diff
@@ -278,7 +273,7 @@
       "profile:sensitive",
       "severity:HIGH"
     ],
-    "alpha.unix.SimpleStream": [
+    "unix.SimpleStream": [
```
Comment (dkrupp) [inline config/labels/analyzers/clangsa.json:276]: maybe to profile:default depending how noisy it is

File: config/labels/analyzers/clang-tidy.json [hunk 3]
```diff
@@ -643,6 +663,10 @@
       "profile:sensitive",
       "severity:MEDIUM"
     ],
+    "bugprone-tagged-union-member-count": [
```
[the remark this answers is not in the document]
Comment (gamesh411) [reply config/labels/analyzers/clang-tidy.json:666]: IMO, it's not particularly easy to understand results, but it's not noisy either.

File: config/labels/analyzers/clang-tidy.json [hunk 4]
```diff
@@ -487,6 +503,10 @@
       "sei-cert:dcl51-cpp",
       "severity:LOW"
     ],
+    "bugprone-return-const-ref-from-parameter": [
+      "doc_url:https://clang.llvm.org/extra/clang-tidy/checks/bugprone/return-const-ref-from-parameter.html",
+      "severity:MEDIUM"
```
[the remark this answers is not in the document]
Comment (gamesh411) [reply config/labels/analyzers/clang-tidy.json:508]: 60 reports, I would consider this moderately noisy.

File: config/labels/analyzers/clang-tidy.json [hunk 5]
```diff
@@ -788,6 +812,13 @@
       "profile:sensitive",
       "severity:MEDIUM"
     ],
+    "cert-arr39-c": [
```
[the remark this answers is not in the document]
Comment (gamesh411) [reply config/labels/analyzers/clang-tidy.json:815]: moving there, this and the bugprone-sizeof-expression is also quite noisy

File: config/labels/analyzers/clang-tidy.json (hunk 1 above)
Comment (gamesh411) [reply config/labels/analyzers/clang-tidy.json:201]: There are 30 reports, we can enable this.

**Comment by gamesh411** [comment]
i have checked the reports for clangsa and tidy, and this is what I came up with

File: config/labels/analyzers/clang-tidy.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (dkrupp) [reply config/labels/analyzers/clang-tidy.json:680]: I think with the documentation it is understandable: https://clang.llvm.org/extra/clang-tidy/checks/bugprone/tagged-union-member-count.html

we could enable this by default I think.

**Review by dkrupp — CHANGES_REQUESTED** [review]
Please add back the severity and link documentation of 
 + cert-arr39-c
 + cert-ctr56-cpp
 
 checkers, but don't assign them to profiles.

I am hesitant about adding bugprone-return-const-ref-from-parameter and bugprone-pointer-arithmetic-on-polymorphic-object to the default profile.

**Review by dkrupp — APPROVED** [review]
LGTM

**Pull request merged**